src/System.IO.FileSystem/src/System/IO/UnixFileSystem.cs

```diff
@@ -62,18 +62,17 @@ public override void MoveFile(string sourceFullPath, string destFullPath)
                 throw new IOException(SR.GetResourceString(SR.IO_AlreadyExists_Name, destFullPath));
             while (Interop.libc.rename(sourceFullPath, destFullPath) < 0)
             {
-                int errno = Marshal.GetLastWin32Error();
-                if (errno == Interop.Errors.EINTR) // interrupted; try again
+                Interop.ErrorInfo errorInfo = Interop.System.GetLastErrorInfo();
+                if (errorInfo.Error == Interop.Error.EINTR) // interrupted; try again
                 {
                     continue;
                 }
-                else if (errno == Interop.Errors.EXDEV) // link fails across devices / mount points
+                else if (errorInfo.Error == Interop.Error.EXDEV) // rename fails across devices / mount points
                 {
                     CopyFile(sourceFullPath, destFullPath, overwrite: false);
                     break;
                 }
-                //else if (errno == Interop.Errors.ENOENT && (!Directory.Exists(Path.GetDirectoryName(sourceFullPath)) || !Directory.Exists(Path.GetDirectoryName(destFullPath)))) // The parent directory for either source or dest can't be found
```

Comment:
I commented on the same thing at https://github.com/dotnet/corefx/pull/2588#discussion-diff-36039386 on @ianhays's file system changes.  Please coordinate with him.